lib/foreman/renderer.rb

```diff
@@ -107,11 +119,10 @@ def indent(count)
 
     # accepts either template object or plain string
     def unattended_render(template, template_name = nil)
-      template_name ||= template.respond_to?(:name) ? template.name : 'Unnamed'
-      raise ::Foreman::Exception.new(N_("Template '%s' is either missing or has an invalid organization or location"), template_name) if template.nil?
+      @template_name ||= template.respond_to?(:name) ? template.name : (template_name || 'Unnamed')
```

Comment:
Because we use it in

    def template_name
      @template_name || 'Unnamed'
    end

Template name was broken and this fixes it. When you rendered a snippet, the name was overriden by the snippet forever. Now it works fine.